src/platform/common/platform/fileSystem.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+
+import * as vscode from 'vscode';
+import { areStringPathsSame } from './fileUtils';
+
+const ENCODING = 'utf8';
+
+export async function writeFile(uri: vscode.Uri, text: string | Buffer): Promise<void> {
```

Comment:
Can we try to use what we have today, I think we should be able to use the IFileSystem as is.
@rchiodo pinging you to check whether the filesystem code has been web-enabled or not.

Reply:
`fileSystem.node.ts` loads `fs-extra` and quite a few `*.node` files so it would be a huge change to make `fileSystem` web enabled in this PR.

Reply:
Sorry, I meant re-using the same interface fofr IFileSystem, yes fs-extra cannot be used.
I was under the impression we already created a version of IFileSystem for web, sounds like we don't have it.